runtime/dynstats.c

```diff
@@ -621,6 +866,11 @@ dynstats_inc(dynstats_bucket_t *b, uchar* metric) {
 			STATSCOUNTER_INC(b->ctrOpsOverflow, b->mutCtrOpsOverflow);
 		}
 	}
+
+	if(b->persistStateInterval > 0 && ++b->nUpdates >= b->persistStateInterval) {
```

Comment:
iirc, this can get called from multiple threads. I think you want to lock the bucker for processing increment and comparision.

Reply:
Also, `persistStateInterval` is being compared to a counter here. This can lead to a cascading failure (is a DoS attack vector) when there is sudden surge in log-messages that causes several writes going in back to back, degrading file-IO in general, potentially leading to back-pressure on other buffered-writes due to dirty-page threshold being hit.

I'd base persist on time (time-interval) not event-count.

If it must be based on event-count, I'd atleast introduce a min-time threshold such that persist is never issued within the threshold-interval of previous persist.